### PR TITLE
Generated QR from genericobject

### PR DESCRIPTION
**Pull request by xacobofg**
Added the possibility of generating a QR from Generic Object #33

### CODE REVIEW

File: inc/qrcode.class.php [hunk 1]
```diff
@@ -85,7 +85,20 @@ function generateQRcode($itemtype, $items_id, $rand, $number, $data) {
...
+            foreach (PluginGenericobjectType::getTypes() as $tmp => $value) {
+               if ($value['itemtype']==$itemtype) {
+                  $a_content[] = 'URL = '.$CFG_GLPI["url_base"]."/plugins/genericobject/front/object.form.php?itemtype=".$itemtype."&id=".$items_id;
+                  $normal=false;
+                  break;
+               }
+            }
+         }
+         if ($normal) {
+            $a_content[] = 'URL = '.$CFG_GLPI["url_base"].Toolbox::getItemTypeFormURL($itemtype, false)."?id=".$items_id;
```
Comment (cedric-anne) [inline inc/qrcode.class.php]: Maybe using `$itemtype::getFormURLWithID($items_id)` would prevent adding all this logic. I think generic object are correctly generating their own URL when using this function.

Comment (xacobofg) [reply inc/qrcode.class.php]: I have tried with the recommended function and it works correctly with generic object

**Comment by cedric-anne** [comment]
Hi,

This looks correct but, reading your PR, I found a more generic solution that is based on item fields. I made a PR with this new way of handling massive actions: #39 .

Can you take a look and make some tests ?

Regards

**Comment by cedric-anne** [comment]
Hi @xacobofg ,

I think you can close this PR as #39 has been merged.

Regards

**Pull request closed without merging**